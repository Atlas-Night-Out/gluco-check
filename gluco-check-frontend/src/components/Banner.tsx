import { useTranslation } from "react-i18next";
import { Button } from "@material-ui/core";

import { JOIN_BETA } from "../lib/constants";
import { Alert, AlertTitle } from "@material-ui/lab";

function Banner() {
  const { t, i18n } = useTranslation();
  const currentLngName = t(
    `languageSelector.availableLanguageLabels.${i18n.language}`
  );
  function joinBeta() {
    window.location.href = `${JOIN_BETA}/${i18n.language}`;
  }

  return (
    <Alert
      variant="standard"
      severity="warning"
      action={
        <Button variant="contained" size="small" onClick={() => joinBeta()}>
          {t("settings.betaBanner.action")}
        </Button>
      }
    >
      <AlertTitle>{t("settings.betaBanner.title")}</AlertTitle>
      {t("settings.betaBanner.message", { language: currentLngName })}
    </Alert>
  );
}

export default Banner;
