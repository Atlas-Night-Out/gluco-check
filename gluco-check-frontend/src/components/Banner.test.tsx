import { render } from "@testing-library/react";
import Banner from "./Banner";

jest.mock("react-i18next", () => ({
  useTranslation: () => {
    return {
      i18n: {
        language: 'debug'
      },
      t: jest.fn().mockImplementation((i) => {
        return i;
      }),
    };
  },
  Trans: function (props: any) {
    return <span>{props.i18nKey}</span>;
  },
}));

describe("Banner component", () => {
  it("renders", async () => {
    const { container } = render(<Banner />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
