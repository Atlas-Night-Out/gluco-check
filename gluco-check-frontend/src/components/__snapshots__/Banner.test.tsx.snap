// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders the component 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-flex-start"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
            />
          </svg>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <p
            class="MuiTypography-root makeStyles-title-1 MuiTypography-body1 MuiTypography-colorPrimary"
          >
            settings.betaBanner.title
          </p>
          <p
            class="MuiTypography-root makeStyles-subtitle-2 MuiTypography-body1"
          >
            settings.betaBanner.message
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-8 MuiGrid-justify-xs-flex-end"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-3 MuiButton-textPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              settings.betaBanner.action
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
